src/ProfessionalGraveyard.tsx

```diff
@@ -7,32 +7,76 @@ import { shortenAddress } from './Utilities';
 
 import Burned from './img/statistics-icons/Burned.png';
 
+import { Trait, Attribute } from './Types';
+
+import { Filter, Traits } from './Filter';
+
 export interface GraveyardProps {
     burntSlugs: BurntSlug[];
+    traitNameMap?: Map<string, Trait>;
+    attributes?: Attribute[];
 }
 
 export function ProfessionalGraveyard(props: GraveyardProps) {
     const {
         burntSlugs,
+        traitNameMap,
+        attributes,
     } = props;
 
-    const data = React.useMemo(() => {
+    const [showFilter, setShowFilter] = React.useState<boolean>();
+
+    const handleToggleFilter = () => {
+        setShowFilter((val) => !val);
+    }
+
+    const [dataFromFilter, setDataFromFilter] = React.useState<Traits>();
+
+    const handleDataFromFilter = React.useCallback((data: Traits) => {
+        setDataFromFilter(data);
+    }, []);
+
+    const filteredSlugs = React.useMemo(() => {
+        if (!dataFromFilter) { 
+            return burntSlugs;
+        }
+
+        const keys = Object.keys(dataFromFilter) as Array<keyof Traits>;
+        return burntSlugs.filter((slug) =>
+            keys.every((key) =>
+                !dataFromFilter[key] ||
+                    slug.attributes.some((attr) =>
+                        attr.trait_type === key && attr.value === dataFromFilter[key]
+                    )
+            )
+        );
+    }, [burntSlugs, dataFromFilter]);
+
+    const slugElements = React.useMemo(() => {
+        if (filteredSlugs.length === 0) {
+            return (
+                <div className="flex items-center justify-center mt-8">
+                    <span className="text-6xl">
```

Comment:
```suggestion
                    <span className="text-4xl text-center">
```

Make smaller and centre